verilog/analysis/checkers/always_ff_non_blocking_rule.h

```diff
@@ -36,19 +37,47 @@ class AlwaysFFNonBlockingRule : public verible::SyntaxTreeLintRule {
   // helper flag or markdown depending on the parameter type.
   static std::string GetDescription(DescriptionType);
 
+  absl::Status Configure(const absl::string_view configuration) override;
   void HandleSymbol(const verible::Symbol& symbol,
                     const verible::SyntaxTreeContext& context) override;
 
   verible::LintRuleStatus Report() const override;
 
+ private:
+  // Detects entering and leaving relevant code inside always_ff
+  bool InsideBlock(const verible::Symbol& symbol, const int depth);
+  // Processes local declarations
+  bool LocalDeclaration(const verible::Symbol& symbol);
+
  private:
   // Link to style guide rule.
   static const char kTopic[];
 
   // Diagnostic message.
   static const char kMessage[];
 
+  // Collected violations.
   std::set<verible::LintViolation> violations_;
+
+  //- Configuration ---------------------
+  bool catch_modifying_assignments_ = false;
+  bool waive_for_locals_ = false;
+
+  //- Processing State ------------------
+  // Inside an always_ff block
+  int inside_ = 0;
+
+  // Stack of inner begin-end scopes
+  struct Scope {
+    int syntax_tree_depth;
+    int inherited_local_count;
```

Comment:
We are only assigning from sources of `size_t` (`locals_.size()` below) so let's that type.

```suggestion
    size_t inherited_local_count;
```